codalab/worker/docker_utils.py

```diff
@@ -256,3 +257,25 @@ def check_finished(container):
             failure_msg = 'Memory limit exceeded.'
         return (True, exitcode, failure_msg)
     return (False, None, None)
+
+
+@wrap_exception('Unable to check Docker container running time')
+def get_container_running_time(container):
+    # Get the current container
+    container = client.containers.get(container.id)
+    # Load this container from the server again and update attrs with the new data.
+    container.reload()
+    # Calculate the start_time of the current container
+    start_time = container.attrs['State']['StartedAt']
+    # Calculate the end_time of the current container. If 'Status' of the current container is not 'exited',
+    # then using the current time as end_time
+    end_time = (
+        container.attrs['State']['FinishedAt']
+        if container.attrs['State']['Status'] == 'exited'
+        else datetime.datetime.now().replace(tzinfo=tz.tzutc()).__str__()
```

Comment:
At least do `str(...)` rather than `__str__`.  But it's better to convert `start_time` and `end_time` once from strings into `datetime` rather than converting to `str` and back.

Reply:
Also, why not `datetime.datetime.utcnow()`?

Reply:
Because I want to exactly match the format of `end_time` with `start_time` when we parse them using `parser.isoparse()`. See the following example:
1. `start_time` is copied directly from attribute `container.attrs['State']['StartedAt']`. When we parse the `start_time` using `parser.isoparse`, there is the `tzinfo` in the return value.
```
>>> start = "2019-11-26T01:15:54.7762612Z"
>>> parser.isoparse(start)
datetime.datetime(2019, 11, 26, 1, 15, 54, 776261, tzinfo=tzutc())
```
2. However, if we use `datetime.datetime.utcnow()` without specifing `tzinfo`, there is no `tzinfo` attribute return in the datetime object.
```
>>> parser.isoparse(str(datetime.datetime.utcnow().isoformat()))
datetime.datetime(2019, 11, 26, 5, 38, 38, 745842)
```
3. When we explicitly specified `tzinfo` in `datetime` object, we also get the `tzinfo` in the return result.
```
>>> end=str(datetime.datetime.now().replace(tzinfo=tz.tzutc()))
>>> parser.isoparse(end)
datetime.datetime(2019, 11, 26, 5, 36, 41, 91737, tzinfo=tzutc())
```
Therefore, I decided to chose `str(datetime.datetime.now().replace(tzinfo=tz.tzutc()))` instead of `datetime.datetime.utcnow()`. Let me know if we want to swith over.

Reply:
Ok, that's fine.